src/main/java/com/znsio/rpap/pages/BasePage.java

```diff
@@ -0,0 +1,81 @@
+package com.znsio.rpap.pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class BasePage extends Page {
+
+    public BasePage(WebDriver driver, WebDriverWait wait) {
+        super(driver, wait);
+    }
+
+    public void inputDataToElement(By by, String dataToInput) {
+
+        WebElement ele = driver.findElement(by);
+        String tagName = ele.getTagName();
+
+        if (tagName.equals("select")) {
```

Comment:
very complex method. Avoid nested if-else

Reply:
Agreed. Keeping this as an improvement task for Raghav/Megha. Added to **Upcoming Changes** section

Reply:
add a TODO in that case so it is easy to find

Reply:
Added